batch/batch/server/database.py

```diff
@@ -0,0 +1,87 @@
+import json
+import uuid
+import asyncio
+import aiomysql
+from asyncinit import asyncinit
+
+
+def run_synchronous(coro):
+    loop = asyncio.get_event_loop()
+    return loop.run_until_complete(coro)
+
+
+@asyncinit
+class Database:
+    @staticmethod
+    def create_synchronous(config_file):
+        db = run_synchronous(Database(config_file))
+        return db
+
+    async def __init__(self, config_file):
+        with open(config_file, 'r') as f:
+            config = json.loads(f.read().strip())
+
+        self.host = config['host']
+        self.port = config['port']
+        self.user = config['user']
+        self.db = config['db']
+        self.password = config['password']
+        self.charset = 'utf8'
+
+        self.pool = await aiomysql.create_pool(host=self.host,
+                                               port=self.port,
+                                               db=self.db,
+                                               user=self.user,
+                                               password=self.password,
+                                               charset=self.charset,
+                                               cursorclass=aiomysql.cursors.DictCursor,
+                                               autocommit=True)
+
+    async def temp_table_name(self, root):
+        suffix = uuid.uuid4().hex[:8]
+        name = f'{root}-{suffix}'
+        niter = 0
+        while await self.has_table(name):
+            suffix = uuid.uuid4().hex[:8]
+            name = f'{root}-{suffix}'
+            niter += 1
+            if niter > 5:
+                raise Exception("Too many attempts to get unique temp table.")
+        return name
+
+    def temp_table_name_sync(self, root):
+        return run_synchronous(self.temp_table_name(root))
+
+    async def has_table(self, name):
+        async with self.pool.acquire() as conn:
+            async with conn.cursor() as cursor:
+                sql = f"SELECT * FROM INFORMATION_SCHEMA.tables " \
+                    f"WHERE TABLE_SCHEMA = %s AND TABLE_NAME = %s"
+                await cursor.execute(sql, (self.db, name))
+                result = cursor.fetchone()
+        return result.result() is not None
```

Comment:
I don't think you actually want this. Any paired use of this with create_table has a race condition. I think the right thing to do is to create a table with a random name catch any errors and try again until you succeed.

Reply:
I don't see the race condition, but I'll rewrite the temporary table creation with a try /except. I think has_table is still a good method to have.

Reply:
I think I see what you mean with the race condition, but the issue is I use these functions in the `test-locally.sh` file to get the table names that are not actually created until the server starts running. I didn't want to have to specify the schema and keys in the `test-locally` file to actually create the table. I can try having the server create the entire batch database and print those temp names. But then I have to figure out how to parse an array output from a Python command in Bash.

Reply:
Ok. I fixed the race condition, but had to take out the test command from `test-locally`. I'll deal with that in a later PR.